Sources/PackageGraph/PackageModel+Extensions.swift

```diff
@@ -24,12 +25,13 @@ extension PackageDependencyDescription {
         //        as its identity, as it will be needed for supporting package
         //        registries.
         let identity = PackageIdentity(url: effectiveURL)
-        
-        return PackageReference(
-            identity: identity,
-            kind: requirement == .localPackage ? .local : .remote,
-            location: effectiveURL
-        )
+
+        if requirement == .localPackage {
+            let path = AbsolutePath(effectiveURL)
+            return PackageReference.local(path: path).with(alternateIdentity: identity)
```

Comment:
as we make progress on the identity changes my goal is to deprecate the idea of "alternate identity". as far as I can tell, the original  goal of the "alternate identity" feature (it had a different name prior to https://github.com/apple/swift-package-manager/pull/3152) was to support an identity as driven by the name field in the manifest which is something that is too free-form to serve as an identity and we should deprecate. as such, my plan was to keep "alternate identity" for backwards compatibility for a while with a deprecation warning on the name field for manifest with new tools-version and restructure the code so its easy to remove when we are ready.

Reply:
to this end, we may want to consider a 'PackageReference.local(identity: , path: )' etc

Reply:
Deprecating this would be great — using the name causes so much confusion.  I think it would make sense to have `PackageReference.local(identity: PackageIdentity? = nil, path: AbsolutePath)`.